### PR TITLE
book name in context

### PR DESCRIPTION
**Pull request by isengrims**

### CODE REVIEW

File: src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/Microservice.kt [hunk 1]
```diff
@@ -65,6 +66,7 @@ class Microservice(
     private val handlerFactory: IHandlerFactory,
     private val manglerFactory: IManglerFactory,
     private val grpcRouter: GrpcRouter,
+    private val bookName: String,
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/Microservice.kt:69]: You can obtain `bookName` from `rootEventId`

File: src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/IHandlerContext.kt [hunk 2]
```diff
@@ -33,6 +33,11 @@ interface IHandlerContext {
      */
     val settings: IHandlerSettings
 
+    /**
+     * Returns book name for this service
+     */
+    val bookName: String
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/IHandlerContext.kt:39]: Why do we need it here?

Comment (isengrims) [reply src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/IHandlerContext.kt:39]: Where is the better place for this? We need to access that from handler when there is data-provider communication.

Comment (ivandruzhinin) [reply src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/IHandlerContext.kt:39]: > We need to access that from handler when there is data-provider communication

That's what I thought, thanks. Ok, let's keep it here

File: src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/Microservice.kt [hunk 3]
```diff
@@ -34,6 +34,7 @@ import com.exactpro.th2.common.schema.message.MessageRouter
 import com.exactpro.th2.common.schema.message.QueueAttribute
 import com.exactpro.th2.common.schema.message.QueueAttribute.EVENT
 import com.exactpro.th2.common.schema.message.QueueAttribute.PUBLISH
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.BookName
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/Microservice.kt:37]: Please remove this import

**Review by ivandruzhinin — CHANGES_REQUESTED** [review]

**Review by ivandruzhinin — APPROVED** [review]

**Pull request merged**